src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -523,6 +523,14 @@ theorem getElem?_zeroExtend (m : Nat) (x : BitVec n) (i : Nat) :
   all_goals (first | apply getLsbD_ge | apply Eq.symm; apply getLsbD_ge)
     <;> omega
 
+theorem getElem_truncate (m : Nat) (x : BitVec n) (i : Nat) (hi : i < m) :
+    (truncate m x)[i] = x.getLsbD i := by
+  simp [getElem?_eq, getElem_zeroExtend]
+
+theorem getElem?_truncate (m : Nat) (x : BitVec n) (i : Nat) :
+    (truncate m x)[i]? = if i < m then some (x.getLsbD i) else none :=
```

Comment:
```suggestion
    (truncate m x)[i]? = if h : i < m then some x[i] else none :=
```

Reply:
I am not sure why you suggest adding `h :` here. AFAIU the hypothesis is taken into account even without the `h`. What is missing here is `h : i < n` to allow for the use of getElem. I moved this to `getElem` but at the cost of the new `h` hypothesis.

Reply:
I'm sorry, I'm wrong again here...